app/services/trainees/create_from_apply.rb

```diff
@@ -43,14 +38,31 @@ def mapped_attributes
         email: raw_contact_details["email"],
         course_code: course&.code,
         training_route: course&.route,
-        degrees: degrees,
         disabilities: disabilities,
-        nationalities: nationalities,
       }.merge(address)
     end
 
-    def degrees
-      raw_degrees.map { |degree| ::Degrees::CreateFromApply.call(attributes: degree) }
+    def personal_details_params
+      {
+        nationality_names: nationality_names,
+      }
+    end
+
+    def create_degrees!
+      ::Degrees::CreateFromApply.call(trainee: trainee)
+    end
+
+    def personal_details_form
+      @personal_details_form ||= PersonalDetailsForm.new(trainee, params: personal_details_params)
+    end
+
+    def save_personal_details!
+      personal_details_form.save!
+    end
+
+    def save_invalid_data!
+      invalid_nationalities = nationality_names - trainee.nationalities.pluck(:name)
```

Comment:
Two things:

1. If there are no invalid nationalities, will this add `{ nationalities: [] }` to the invalid data json? Do we want a nationality key if there are no invalid nationalities?
2. In what case could a nationality not be valid? Is it because we don't have have a mapping for it in `ApplyApi::CodeSets::Nationalities::MAPPING`? In that case, `nationality_names` would return `[nil]` and we would add `{ nationalities: [nil] }` to the invalid data json, which isn't so helpful.

Reply:
We have decided to get rid of the invalid data management for nationalities altogether, since from the codesets, it seems like we have a very good mapping from DTTP->Register->Apply nationalities.

However, in the unlikely event of a nationality being unmappable due to register not having any of (new) Apply values, added a capture to Sentry.